packages/react-search-ui/src/helpers.ts

```diff
@@ -1,6 +1,88 @@
+import { FacetValue, FilterValue, FilterType } from "@elastic/search-ui";
+import {
+  FacetDefaultOptionProps,
+  BeaconFacetValuesMapProps
+} from "@elastic/react-search-ui-views";
+
 // LÒpez => Lopez
 // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize
 export const accentFold = (str: any): string =>
   typeof str === "string"
     ? str.normalize("NFD").replace(/[\u0300-\u036f]/g, "")
     : "";
+
+export function getValueToSearchFromFacetValue(facetValue: FacetValue): string {
+  switch (typeof facetValue.value) {
+    case "string":
+      return accentFold(facetValue.value).toLowerCase();
+    case "number":
+      return facetValue.value.toString();
+    case "object": {
+      return "name" in facetValue.value
+        ? accentFold(facetValue.value.name).toLowerCase()
+        : "";
+    }
+    default:
+      return "";
+  }
+}
+
+export function isMatchedFacetValueAndSearchTerm(
+  facetValue: FacetValue,
+  searchTerm: string
+) {
+  const valueToSearch = getValueToSearchFromFacetValue(facetValue);
+  return valueToSearch.includes(accentFold(searchTerm).toLowerCase());
+}
+
+export function isMatchedDefaultOptionAndSearchTerm(
+  option: FacetDefaultOptionProps,
+  searchTerm: string
+) {
+  const valueToSearch =
+    typeof option === "string"
+      ? accentFold(option).toLowerCase()
+      : accentFold(option.value).toLowerCase();
+
+  return valueToSearch.includes(accentFold(searchTerm).toLowerCase());
+}
+
+/* 
+Stringify the filterValue which can be a string, number, boolean, object with a name property,
+or an array of string, number, or boolean. This is helpful for building keys to be used in hashmaps
+of facetValues.
+*/
+export const stringifyFilterValue = (filterValue: FilterValue): string => {
```

Comment:
Why can't we just use `JSON.stringify` here?

Reply:
The `FilterValue` type is a union of `FilterValueValue` and `FilterValueRange`
https://github.com/Tsquare/search-ui/blob/wxu/beacon-facet-2/packages/search-ui/src/types/index.ts#L28

The `FilterValueRange` is an object with a key called `name` and that's the stringified value that we should be using.

As a result, this utility function does some type checking to be able to return that name key in that situation

Reply:
On a related note, I later realized that Elastic also has a similar utility function called `getFilterValueDisplay` in another package of `react-search-ui-views`
https://github.com/Tsquare/search-ui/blob/wxu/beacon-facet-2/packages/react-search-ui-views/src/view-helpers/getFilterValueDisplay.ts#L6C11-L6C11

I thought about perhaps removing this utility function and using that instead to stay DRY but also had some concern about Elastic possibly changing that function in the future and breaking this use case within BeaconFacet.

I felt a little bit of code duplication doesn't hurt to avoid that risk. But curious to get your thoughts